KDOP_tree/include/CGAL/KDOP_tree/KDOP_kdop.h

```diff
@@ -0,0 +1,185 @@
+    double min_height() const { return *std::min_element( array_height_, array_height_ + direction_number ); }
+
+    /// Inline function to return the maximum support height.
+    double max_height() const { return *std::max_element( array_height_, array_height_ + direction_number ); }
+
+#ifdef DEBUG_
+    /// Inline function to set directions
+    void set_directions(const Vec_direction& vec_direction) { vector_direction_.clear(); vector_direction_ = vec_direction; }
+
+    /// Inline function to return directions.
+    Vec_direction directions() const& { return vector_direction_; }
+
+    /*!
+     * @brief Add a new direction to existing directions.
+     * @param new_direction the new direction to be added
+     * @return updated vector of (k + 1) directions
+     */
+    void add_direction(Direction_type new_direction) { vector_direction_.push_back(new_direction); }
+#endif
+
+    /*!
+     * @brief Check if two k-dops overlap by comparing support heights of
+     * the two k-dops.
+     * @param kdop1 the k-dop of the query
+     * @return true if the two k-dops overlap; otherwise, false.
+     * \todo Add the checking function.
+     */
+    bool do_overlap(const Kdop& kdop) const;
+
+  private:
+    Vec_direction vector_direction_; // Redundant at the moment
+
+    std::array<double, direction_number> array_height_;
+
+  };
+
+  template<typename GeomTraits, unsigned N>
+  void KDOP_kdop<GeomTraits, N>::compute_support_heights(const Vec_direction& directions, const Triangle_3& t)
+  {
+    int num_directions = directions.size();
+
+    for (int i = 0; i < num_directions; ++i) { // number of directions
+      Point_3 direction = directions[i];
+
+      std::cout << "direction: " << direction.x() << ", " << direction.y() << ", " << direction.z() << std::endl;
+
+      std::vector<double> heights;
+      for (int j = 0; j < 3; ++j) { // number of vertices
+        Point_3 v = t.vertex(j);
+        double height = v.x()*direction.x() + v.y()*direction.y() + v.z()*direction.z();
+        heights.push_back(height);
+
+        std::cout << "vertex: " << v.x() << ", " << v.y() << ", " << v.z() << ": height = " << height << std::endl;
+      }
+
+      double height_max = *std::max_element(heights.begin(), heights.end());
+
+      array_height_[i] = height_max;
+    }
+  }
+
+  template<typename GeomTraits, unsigned int N>
+  bool KDOP_kdop<GeomTraits, N>::do_overlap(const Kdop& kdop_query) const
+  {
+    bool is_overlap = true;
+
+    std::array<double, direction_number> support_heights = this->support_heights();
+    std::array<double, direction_number> support_heights_query = kdop_query.support_heights();
+
+    int num_support_heights = direction_number;
+
+    int num_non_overlap = 0;
+    for (int i = 0; i < num_support_heights; ++i) {
+      double height1 = support_heights[i];
+      double height2 = support_heights_query[i];
+
+      if (height1 + height2 < 0) {
+        num_non_overlap += 1;
```

Comment:
You could write a `return` directly here, instead of incrementing, breaking, testing.